accepted/2021/net6.0-tfms/net6.0-tfms.md

```diff
@@ -0,0 +1,108 @@
+
+### Non-Goals
+
+* Re-design the TFM syntax
+
+## Stakeholders and Reviewers
+
+* MSBuild / SDK team
+* NuGet team
+* Project system team
+* Xamarin team
+
+## Design
+
+| TFM                | Compatible With                          |
+|--------------------|------------------------------------------|
+| net6.0             | (subsequent version of net5.0)           |
+| net6.0-windows     | (subsequent version of net5.0-windows)   |
+| net6.0-android     | xamarin.android                          |
+|                    | (+everything else inherited from net6.0) |
+| net6.0-ios         | xamarin.ios                              |
+|                    | (+everything else inherited from net6.0) |
+| net6.0-macos       | xamarin.mac                              |
+|                    | (+everything else inherited from net6.0) |
+| net6.0-maccatalyst | xamarin.ios                              |
+|                    | (+everything else inherited from net6.0) |
+| net6.0-tvos        | xamarin.tvos                             |
+|                    | (+everything else inherited from net6.0) |
+| net6.0-watchos     | xamarin.watchos                          |
+|                    | (+everything else inherited from net6.0) |
+
+### Compatibility rules
+
+The [built-in compatibility rules][net5.0] of the `netX.Y-platform` TFMs make a
+TFM automatically compatible with:
+
+1. Itself
+2. Earlier versions of itself
+3. The corresponding `netX.Y` TFM
+4. Earlier versions of the corresponding `netX.Y` TFM
+
+The `net5.0` TFM has the additional twist that it's compatible with
+`netcoreapp3.2` (and earlier), `netstandard2.1` (and earlier) as well as .NET
+Framework (`net4.x` and earlier, but under a warning). The .NET Framework
+compatibility is achieved as a *fallback*, that is, it's only used when nothing
+else applies.
+
+We don't want to model the Xamarin TFMs as a fallback but instead want them to
+apply **before** `netcoreapp` and `netstandard`. Usually the existence of the
+Xamarin TFM implementation is to bait a `netstandard` API and switch the
+platform specific implementation. Most of the Plugin type packages use this
+pattern and in fact the `netstandard` "implementation" is more like a reference
+assembly with just stubs that throw at runtime. `Xamarin.Essentials` is a good
+example of this. The same rational can be applied to `netcoreapp` where this
+would logically be the server implementation (or maybe a .NET Core 3.x desktop
+implementation).
+
+> ***OPEN QUESTION**: Should `net6.0-maccatalyst` also have a compatibility
+> relationship with any of the **existing** Xamarin TFMs, such as `xamarin.mac`
+> or `xamarin.tvos`?*
+>
+> ***OPEN QUESTION**: Should `net6.0-maccatalyst` also have a compatibility
+> relationship with any of the **new** Xamarin TFMs, such as `net6.0-mac`?*
+>
+> ***OPEN QUESTION**: Should `net6.0-maccatalyst` generate a warning when assets
+> from `xamarin.ios` are being used?
```

Comment:
@Redth @rolfbjarne @marek-safar please help me answer these.

Reply:
I think that warnings would be useful any time assets are consumed in the `net6.0-maccatalyst` TFM from `net6.0-ios`, `xamarin.ios10`, and if we add support: `net6.0-mac` and `xamarin.mac20`. 

A couple of reasons:
1. There may of course be code that throws at runtime (some iOS API's do not work in Catalyst and some AppKit/Cocoa API's do not work as well).
2. Bindings to native frameworks/libraries generally will not work (the native assets themselves would need to be rebuilt for Catalyst).
3. MSBuild logic, especially in the older, non-net6 TFM's may not work correctly.

Reply:
It's also probably worth explicitly stating that `net6.0-maccatalyst` does not have compatibility with `net6.0-ios` or `net6.0-mac` assets.  I had originally assumed we might add this, but I see you did not include that and realize that was likely quite intentional, and a better choice the more I think about it.

Reply:
> ***OPEN QUESTION**: Should `net6.0-maccatalyst` also have a compatibility
> relationship with any of the **existing** Xamarin TFMs, such as `xamarin.mac`
> or `xamarin.tvos`?*

`xamarin.ios` - yes (as you have in the table in the spec)

`xamarin.mac/watchOS/tvos` - no:

I would guess most NuGets in the ecosystem that would have `xamarin.mac` assets would already have `xamarin.ios` assets as well.  Also, Catalyst supports a large subset of the iOS API's, but a smaller subset off the Mac (AppKit/Cocoa) API's (at least officially).  I don't think there's enough value in adding compatibility for `Xamarin.mac` assets vs the potential incompatibilities in that scenario.  Same for other Xamarin apple TFM's.

Reply:
> ***OPEN QUESTION**: Should `net6.0-maccatalyst` also have a compatibility
> relationship with any of the **new** Xamarin TFMs, such as `net6.0-mac`?*

No.

I originally thought yes for `net6.0-ios` however I'm not convinced this is the correct answer.

1. If you are building for the new `net6.0-ios` target, it doesn't seem unreasonable to expect to include the `net6.0-catalyst` target if that's the real intent.  I expect library authors will see PR's to add the TFM rather quickly if the code is actually able to be shared with no or minimal changes.
2. Native frameworks/libs need to be rebuilt for `net6.0-maccatalyst` and are slightly different than the native assets you would build for `net6.0-ios` bindings.  These would be incompatible.

No for `net6.0-mac`, `net6.0-tvos`, etc as well.  These would be potentially even less compatible with catalyst as they support a more narrow set of iOS API's.

Reply:
I agree with @Redth on all three questions.

Reply:
Awesome, I have addressed all issues now.